Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm

```diff
@@ -173,6 +174,54 @@
     });
 }
 
+bool WebExtensionContext::shouldDisplayBlockedResourceCountAsBadgeText()
+{
+    return objectForKey<NSNumber>(m_state, displayBlockedResourceCountAsBadgeTextStateKey).boolValue;
+}
+
+void WebExtensionContext::saveShouldDisplayBlockedResourceCountAsBadgeText(bool shouldDisplay)
+{
+    [m_state setObject:@(shouldDisplay) forKey:displayBlockedResourceCountAsBadgeTextStateKey];
+    writeStateToStorage();
+}
+
+void WebExtensionContext::incrementActionCountForTab(WebExtensionTab* tab, ssize_t incrementAmount)
```

Comment:
`WebExtensionTab&` would be best, since tab is required right? if tab is null it will increment the default action, for all tabs, which would be unexpected for this API I think.

Reply:
Tab is required, yes.